nixos/modules/services/web-apps/libretranslate.nix

```
@@ -0,0 +1,197 @@
+        default = "/var/lib/libretranslate";
+        example = "/srv/data/libretranslate";
+        description = lib.mdDoc "The data directory.";
+      };
+      threads = lib.mkOption {
+        type = lib.types.nullOr lib.types.int;
+        default = null;
+        example = 8;
+        description = lib.mdDoc "Set number of threads.";
+      };
+      enableApiKeys = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        example = true;
+        description = lib.mdDoc "Enable API keys database.";
+      };
+      disableWebUI = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        example = true;
+        description = lib.mdDoc "Disable web ui.";
+      };
+      updateModels = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        example = true;
+        description = lib.mdDoc "Update language models at startup";
+      };
+      domain = lib.mkOption {
+        type = lib.types.str;
+        default = "";
+        example = "libretranslate.example.com";
+        description = lib.mdDoc ''
+          The domain serving your LibreTranslate instance.
+          Required for configure nginx as a reverse proxy.
+        '';
+      };
+      configureNginx = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = lib.mdDoc "Configure nginx as a reverse proxy for LibreTranslate.";
+      };
+      extraArgs = lib.mkOption {
+        type = with lib.types; attrsOf (nullOr (oneOf [ bool str int (listOf (oneOf [ bool str int ])) ]));
+        default = { };
+        example = {
+          debug = true;
+          disable-files-translation = true;
+          url-prefix = "translate";
+        };
+        description = lib.mdDoc "Extra arguments passed to the LibreTranslate.";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    environment.systemPackages = [ ltmanageKeysCli ];
+
+    systemd.tmpfiles.rules = lib.mkIf (cfg.dataDir != "/var/lib/libretranslate") [
+      "d '${cfg.dataDir}' 0750 ${cfg.user} ${cfg.group} - -"
+      "z '${cfg.dataDir}' 0750 ${cfg.user} ${cfg.group} - -"
+    ];
+
+    systemd.services.libretranslate = {
+      description = "LibreTranslate service";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      environment = {
+        HOME = cfg.dataDir;
+      };
+      serviceConfig = lib.mkMerge [{
+        Type = "simple";
+        ExecStart = ''
+          ${cfg.package}/bin/libretranslate \
+            '--host' '${cfg.host}' \
+            '--port' '${toString cfg.port}' \
+            ${lib.optionalString cfg.enableApiKeys "'--api-keys'"} \
+            ${lib.optionalString cfg.disableWebUI "'--disable-web-ui'"} \
+            ${lib.optionalString cfg.updateModels "'--update-models'"} \
+            ${lib.optionalString (cfg.threads != null) "'--threads' '${toString cfg.threads}'"} \
+            ${lib.cli.toGNUCommandLineShell { } cfg.extraArgs}
```

Could alternatively write it like this, since we're already using `toGNUCommandLineShell`. Also makes it possible for the user to override the flags if they wish.

```suggestion
          ${cfg.package}/bin/libretranslate ${lib.cli.toGNUCommandLineShell { }
            (cfg.extraArgs // {
              inherit (cfg) host port threads;
              api-keys = cfg.enableApiKeys;
              disable-web-ui = cfg.disableWebUI;
              update-models = cfg.updateModels;       
            })
          }
```

---

This change throws an error:
```
       … while evaluating the attribute 'config.system.build.toplevel'

         at /home/lafiel/works/src-nix/nixpkgs/lib/modules.nix:322:9:

          321|         options = checked options;
          322|         config = checked (removeAttrs config [ "_module" ]);
             |         ^
          323|         _module = checked (config._module);

       … while calling the 'seq' builtin

         at /home/lafiel/works/src-nix/nixpkgs/lib/modules.nix:322:18:

          321|         options = checked options;
          322|         config = checked (removeAttrs config [ "_module" ]);
             |                  ^
          323|         _module = checked (config._module);

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: syntax error, unexpected end of file, expecting IND_STR or DOLLAR_CURLY or IND_STRING_CLOSE

       at /home/lafiel/works/src-nix/nixpkgs/nixos/modules/services/web-apps/libretranslate.nix:184:52:

          183|         locations."^~ /static/" = {
          184|           alias = "${cfg.package.static-compressed}/share/libretranslate/static/";
             |                                                    ^
          185|         };

```

---

Fixed.